vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/logcat/LogcatCollector.kt

```diff
@@ -50,6 +51,10 @@ class LogcatCollector : LogcatEventsListener, LogsProvider {
             is LogcatEvent.BatchStarted -> {
                 devices.compute(event.device) { _, oldState ->
                     val state = oldState ?: DeviceState()
+                    if (state.currentBatchId != null && state.currentBatchId != event.batchId) {
+                        // previous unfinished batch, let's close it
+                        batchCollectors[state.currentBatchId]?.onBatchFinished()
```

Comment:
close previous batch earlier if it wasn't finished for some reason (such as emulator failure), should avoid waiting for timeout in most cases